### PR TITLE
Feature/jcs testing refactor

### LINKED ISSUE #82: Testing for helper functions
Write tests for conversion functions

### LINKED ISSUE #83: Remove any types from Location, weeklyForecast, App
- [ ] Follow it up the chain to remove the type any
- [ ] Look at errors what do they have in common?
- [ ] Knowing what you know now, how would you restructure the TS architecture ?
- [ ] What was the biggest issue we kept running into?

### LINKED ISSUE #84: Adjust conversions to metric
- [ ] in location change conversions to metric
- [ ] in weekly forecast update conversions to metric

### PR DESCRIPTION
**Pull request by JoshSevy**
# Description
Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.
- Refactored TS from the weeklyForecast to App up the chain
- Added in tests

## Type of change
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Other, Please explain:

- Add tests to all helper functions in conversions
- Add edge casing to functions in conversions
- Add metric conversions to milage items
- Add clock back in and applied styling
## Has This Been Tested?
- [ ] No
- [x] Yes
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration:
Ran app for potential bugs, and tested conversions test written using jest
# Message for reviewer:
I think all of the typescript is moved away from using any and have no error through what I worked on. There is some typescript error warnings in home but it just got to late. 

# Checklist:
- [x] My code has no unused/commented out code
- [x] My code has no console logs
- [x] I have performed a self-review of my own code
- [x] I have commented on my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/helpers/conversions.test.tsx [hunk 1]
```diff
@@ -0,0 +1,63 @@
+import React from 'react'
+import '@testing-library/react'
+import { aqIndex, degToDirection, forecastDtDisplay, weatherDtDisplay, weatherIcon, uvIndex, temp} from './conversions'
+
```
Comment (ErinUntermeyer) [inline src/helpers/conversions.test.tsx:4]: Awesome job! Thanks for testing these methods.

**Comment by ErinUntermeyer** [comment]
Looks great Josh!

**Review by ErinUntermeyer — APPROVED** [review]

File: src/Location/Location.tsx [hunk 2]
```diff
@@ -13,9 +13,15 @@ import cough from '../assets/cough.png'
 
 import { degToDirection, weatherDtDisplay, weatherIcon, aqIndex, uvIndex, temp } from '../helpers/conversions'
 
-import { DetailsProps } from '../helpers/detailsdefinitions'
+import { DataTypes } from '../helpers/typeDefinitions'
 
-const Location = (props: any) => {
+interface LocationProps {
```
Comment (leighlars) [inline src/Location/Location.tsx:18]: great job converting all these!

File: src/helpers/apiCalls.tsx [hunk 3]
```diff
@@ -19,7 +19,7 @@ export const getCoordinates = (query: string) => {
 
 export const getWeatherData = async (lat: number, long: number) => {
   return await fetch(
-      `https://api.openweathermap.org/data/2.5/onecall?lat=${lat}&lon=${long}&appid=${weatherDataKey}`
+      `https://api.openweathermap.org/data/2.5/onecall?lat=${lat}&lon=${long}&appid=${weatherDataKey3}`
```
Comment (leighlars) [inline src/helpers/apiCalls.tsx:22]: thanks for this!

File: src/helpers/conversions.test.tsx [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+describe('conversions testing', () => {
+  it('should check if degToDirection it functioning correctly', () => {
+
+  expect(degToDirection(234)).toEqual('SW')
```
Comment (leighlars) [inline src/helpers/conversions.test.tsx:8]: Thank you for testing the conversions!

File: src/helpers/typeDefinitions.ts [hunk 5]
```diff
@@ -0,0 +1,114 @@
+export interface DataTypes {
```
Comment (leighlars) [inline src/helpers/typeDefinitions.ts:1]: thank you for handling the data types

**Review by leighlars — APPROVED** [review]
Great job! Thank you for doing this!

**Pull request merged**